src/parser.rs

```diff
@@ -193,12 +193,11 @@ impl<'a> Parser<'a> {
         }
     }
 
-    pub(crate) fn parse_inner_list(&mut self) -> SFVResult<InnerList> {
+    fn parse_inner_list(&mut self) -> SFVResult<InnerList> {
         // https://httpwg.org/specs/rfc8941.html#parse-innerlist
 
-        if Some(b'(') != self.input.next() {
-            return Err("parse_inner_list: input does not start with '('");
-        }
```

Comment:
As I mentioned, I'd prefer to keep these checks in place.
We can peek the next character to perform the check if you want to maintain the pointer to the failing character in the input.